src/ophyd_async/panda/panda.py

```diff
@@ -33,148 +20,16 @@ class PcapBlock(Device):
     arm: SignalRW[bool]
 
 
-def _block_name_number(block_name: str) -> Tuple[str, Optional[int]]:
-    """Maps a panda block name to a block and number.
-
-    There are exceptions to this rule; some blocks like pcap do not contain numbers.
-    Other blocks may contain numbers and letters, but no numbers at the end.
-
-    Such block names will only return the block name, and not a number.
-
-    If this function returns both a block name and number, it should be instantiated
-    into a device vector."""
-    m = re.match("^([0-9a-z_-]*)([0-9]+)$", block_name)
-    if m is not None:
-        name, num = m.groups()
-        return name, int(num or 1)  # just to pass type checks.
-
-    return block_name, None
-
-
-def _remove_inconsistent_blocks(pvi_info: Optional[Dict[str, PVIEntry]]) -> None:
-    """Remove blocks from pvi information.
-
-    This is needed because some pandas have 'pcap' and 'pcap1' blocks, which are
-    inconsistent with the assumption that pandas should only have a 'pcap' block,
-    for example.
-
-    """
-    if pvi_info is None:
-        return
-    pvi_keys = set(pvi_info.keys())
-    for k in pvi_keys:
-        kn = re.sub(r"\d*$", "", k)
-        if kn and k != kn and kn in pvi_keys:
-            del pvi_info[k]
-
-
-class PandA(Device):
+class CommonPandABlocks(Device):
```

Comment:
Is this abstract or will it ever be instantiated not as the subclass?

Reply:
It is abstract. I should make that explicit probably.

Reply:
I don't think there's a nice way to do this without making methods, which I don't really want to do. I don't think we need to explicitly force it to be abstract.

Reply:
Does this prevent instantiating an instance of CommonPandABlocks if are no @abstractmethods?

```
from abc import ABC

class CommonPandABlocks(Device, ABC):
```

Reply:
I'll step through pvi.py more closely next week but otherwise happy to let that go